### PR TITLE
check in endpoint

### PR DESCRIPTION
**Pull request by romanetar**
ref https://tipit.avaza.com/project/view#!tab=task-pane&groupby=MyTaskDueDate&view=vertical&task=2916710&fileview=grid

Signed-off-by: romanetar <roman_ag@hotmail.com>

### CODE REVIEW

File: app/Http/Controllers/Apis/Protected/Summit/OAuth2SummitBadgeScanApiController.php [hunk 1]
```diff
@@ -409,6 +432,36 @@ function() use($summit) {
         );
     }
 
+    /**
+     * @param $summit_id
+     * @return mixed
+     */
+    protected function checkIn($summit_id) {
+        return $this->processRequest(function () use ($summit_id) {
+            if(!Request::isJson()) return $this->error400();
```
Comment (smarcet) [inline app/Http/Controllers/Apis/Protected/Summit/OAuth2SummitBadgeScanApiController.php:441]: use please trait
 use GetAndValidateJsonPayload like here
https://github.com/OpenStackweb/summit-api/blob/eacdd5ff4655dfb7ba68b0d664d7fa2af0024c88/app/Http/Controllers/Apis/Protected/Summit/OAuth2SummitOrderExtraQuestionTypeApiController.php#L219

File: app/Models/Foundation/Summit/Registration/SummitOrder.php [hunk 2]
```diff
@@ -75,6 +75,8 @@ class SummitOrder extends SilverstripeBaseModel implements IQREntity
      */
     private $owner_first_name;
 
+    public static $company;
```
Comment (smarcet) [inline app/Models/Foundation/Summit/Registration/SummitOrder.php:78]: could we remove this one ?

File: app/Services/Model/AttendeeService.php [hunk 3]
```diff
@@ -469,4 +477,20 @@ public function doVirtualCheckin(Summit $summit, int $attendee_id): ?SummitAtten
             return $attendee;
         });
     }
+
+    public function doCheckIn(Summit $summit, String $qr_code): void
+    {
+        $this->tx_service->transaction(function() use($summit, $qr_code){
+
+            $fields        = SummitAttendeeBadge::parseQRCode($qr_code);
+            $ticket_number = $fields['ticket_number'];
```
Comment (smarcet) [inline app/Services/Model/AttendeeService.php:486]: could we add validations here for the QR prefix ?
https://github.com/OpenStackweb/summit-api/blob/172a04a7407e7e96f7c08653429513ea606e0a7f/app/Services/Model/Imp/SponsorUserInfoGrantService.php#L120

File: app/Models/Foundation/Summit/Registration/SummitOrder.php (hunk 2 above)
Comment (romanetar) [reply app/Models/Foundation/Summit/Registration/SummitOrder.php]: done

File: app/Http/Controllers/Apis/Protected/Summit/OAuth2SummitBadgeScanApiController.php (hunk 1 above)
Comment (romanetar) [reply app/Http/Controllers/Apis/Protected/Summit/OAuth2SummitBadgeScanApiController.php:442]: done

File: app/Services/Model/AttendeeService.php (hunk 3 above)
Comment (romanetar) [reply app/Services/Model/AttendeeService.php:486]: done

File: app/Services/Model/AttendeeService.php [hunk 4]
```diff
@@ -469,4 +477,20 @@ public function doVirtualCheckin(Summit $summit, int $attendee_id): ?SummitAtten
...
+
+            $badge = $this->badge_repository->getBadgeByTicketNumber($ticket_number);
+
+            if(is_null($badge))
+                throw new EntityNotFoundException("badge not found");
+
+            $badge->getTicket()->getOwner()->setSummitHallCheckedIn(true);
```
[the remark this answers is not in the document]
Comment (romanetar) [reply app/Services/Model/AttendeeService.php]: done

[the remark this answers is not in the document]
Comment (romanetar) [reply app/Services/Model/AttendeeService.php]: done

**Review by smarcet — APPROVED** [review]
LGTM

**Pull request merged**